client/state/happychat/test/middleware.js

```diff
@@ -147,12 +147,12 @@ describe( 'middleware', () => {
 
 		it( 'should send relevant browser information to the connection', () => {
 			const getState = () => state;
-			const connection = { info: spy() };
-			const action = { type: HAPPYCHAT_SEND_USER_INFO, siteUrl: 'http://butt.holdings/' };
+			const connection = { sendInfo: spy() };
+			const action = { type: HAPPYCHAT_SEND_USER_INFO, site: { URL: 'http://butt.holdings/' } };
 			sendInfo( connection, { getState }, action );
 
-			expect( connection.info ).to.have.been.calledOnce;
-			expect( connection.info.firstCall.args[ 0 ].text ).to.include( state.happychat.geoLocation.city );
+			expect( connection.sendInfo ).to.have.been.calledOnce;
+			expect( connection.sendInfo.firstCall.args[ 0 ].site.URL ).to.equal( action.site.URL );
```

Comment:
Could we also mock `window`, `screen` and `navigator` global objects in this test so that all its code paths are covered?

In this case, I think we don't need a comprehensive mock like [mock-browser](https://github.com/darrylwest/mock-browser), just need to mock the relevant part, e.g.
```
global.window = {
    innerWidth: ...,
    innerHeight: ...,
};
```
and then tear it down after running the tests.